File: backend/mong-lesson/src/main/java/spring/edu/monglesson/model/Video.java
package spring.edu.monglesson.model;

import java.sql.Time;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Video {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private Time time;
    private String videoSource;
    private String lessonID;

    public Video() {
    }

    public Video(Long id, String name, Time time, String videoSource, String lessonID) {
        this.id = id;
        this.name = name;
        this.time = time;
        this.videoSource = videoSource;
        this.lessonID = lessonID;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Time getTime() {
        return time;
    }

    public void setTime(Time time) {
        this.time = time;
    }

    public String getVideoSource() {
        return videoSource;
    }

    public void setVideoSource(String videoSource) {
        this.videoSource = videoSource;
    }

    public String getLessonID() {
        return lessonID;
    }

    public void setLessonID(String lessonID) {
        this.lessonID = lessonID;
    }

}

File: fontend/app-teacher/src/app/model/video.ts
export class Video {
  id: number;
  name: string;
  time: string; // In TypeScript, you can represent time as a string
  videoSource: string;
  lessonID: string;

  constructor(
    id: number,
    name: string,
    time: string,
    videoSource: string,
    lessonID: string
  ) {
    this.id = id;
    this.name = name;
    this.time = time;
    this.videoSource = videoSource;
    this.lessonID = lessonID;
  }
}
